code/edgeidentity/build.gradle

```diff
@@ -43,6 +47,13 @@ android {
     buildTypes {
         debug {
             testCoverageEnabled true
+            debuggable true
+
+            // Functional tests will fail without this flag as this module exceeds the DEX limit.
+            // It is sufficient to enable this for debug as it is the type used for functional tests.
+            // Consuming apps are responsible for making changes to their gradle to overcome
+            // the DEX limit as necessary.
+            multiDexEnabled true
```

Comment:
I think you can remove this if you remove the dependencies `implementation 'androidx.appcompat:appcompat:1.2.0'` and `androidTestImplementation "androidx.test:core:1.4.0"`. In my testing neither dependency was needed.

However, when I also updated `com.fasterxml.jackson.core:jackson-databind` to 2.12.7 and `androidx.test.espresso:espresso-core` to 3.5.0, then I went over the dex limit again by about 20 methods. At this point I'd suggest removing multidex and the to unneeded dependencies.